### ISSUE TITLE
Window content sometimes not centered in Gnome Wayland session

### ISSUE THREAD

**Issue by Rakksor** [SFML/SFML#2709]
## Subject of the issue

When opening a fullscreen window with SFML in a Gnome Wayland session the window content is sometimes offset by what seems to be the height of the Gnome top bar. Maybe this is related to #1613?

## Your environment

* Arch Linux, Gnome 44.5
* SFML 2.6.0
* g++ 13.2.1

## Steps to reproduce

Run this minimal example in a Gnome Wayland session until the issue appears (might take some tries, maybe appears more often when starting while a Firefox window is visible):

```cpp
#include "SFML/Graphics.hpp"

int main() {
	sf::RenderWindow Window(sf::VideoMode::getDesktopMode(), "Test", sf::Style::Fullscreen);
	sf::Event Event;

	while (Window.isOpen()) {
		while (Window.pollEvent(Event)) {
			if (Event.type == sf::Event::Closed)
				Window.close();
		}
				
		Window.clear(sf::Color(255, 0, 0, 255));
		Window.display();
	}
	return 0;
}
```

## Expected behavior

The window content fills the whole screen. In the above example, this means the whole screen is red. This is also what usually happens:

![expected_result](https://github.com/SFML/SFML/assets/14354807/39ec9d0c-2e46-441c-b464-30c834859973)

## Actual behavior

Sometimes the window content is offset by what seems to be the height of the Gnome top bar and cut off by that amount at the bottom. The above example only shows the first part, the red does not start at the top of the screen but is offset by a grey bar:

![actual_result](https://github.com/SFML/SFML/assets/14354807/1d612e8d-c518-4471-bd3a-2f6fb11c8537)

The grey bar is part of the window as seen in the overview:

![actual_result_overview](https://github.com/SFML/SFML/assets/14354807/9b90fefa-65cc-4882-a426-305ab88d2e7a)

And as a comparison the desktop view showing the top bar:

![desktop](https://github.com/SFML/SFML/assets/14354807/ff1a9032-518e-421c-afd5-1a7206be9c0b)

**Comment by eXpl0it3r** [comment]
As SFML doesn't natively support Wayland, what is used to translate the X11 calls to Wayland?

**Comment by danieljpetersen** [comment]
This may or may not be fixed? I've been on Gnome 46.2 Wayland the past few days and haven't noticed this.

**Comment by eXpl0it3r** [comment]
I'm sure it won't be the last potentially Wayland reported issue, but I'm closing it for now as we have a positive feedback and missing additional information.
